components/banner/countdown.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component, Fragment } from 'react'
+import classNames from 'classnames'
+import { Button } from 'antd';
+import stylesheet from './countdown.scss'
+import { countdownInfo } from '../constants'
+import { InputForm } from '../inputForm/input-form';
+
+const tokensalePhaseStartDate = countdownInfo.finalDate;//month starts with 0
+const millisecondsInOneDay = 86400000;
+const millisecondsInOneHour = 1000 * 60 * 60;
+const millisecondsInOneMinute = 1000 * 60;
+const millisecondsInOneSeconds = 1000;
+
+class Timer extends Component {
+  constructor(){
+    super();
+
+    this.state = { currentCountdown: this.getCountdown() }
+    this.intervalId = null;
+
+  }
+
+  getCountdown = () => {
+    var today = new Date();
+    var difference = tokensalePhaseStartDate.getTime() - today.getTime();
+
+    if(difference < 0){
+      clearInterval(this.intervalId);
+      return;
+    }
+
+    var daysUntil = Math.floor(difference / millisecondsInOneDay);
+
+    var hourMilliseconds = difference - (daysUntil * millisecondsInOneDay);
+    var hours = Math.floor(hourMilliseconds / millisecondsInOneHour);
+
+    var minutesMilliseconds = difference - (daysUntil * millisecondsInOneDay) - (hours *  millisecondsInOneHour);
+    var minutes = Math.floor(minutesMilliseconds / millisecondsInOneMinute);
+
+    var secondsMillisecons = difference - (daysUntil * millisecondsInOneDay) - (hours * millisecondsInOneHour) - (minutes * millisecondsInOneMinute);
+    var seconds = Math.floor(secondsMillisecons / millisecondsInOneSeconds);
+
+    return {days: daysUntil, hours: hours, minutes: minutes, seconds: seconds};
+  }
```

Comment:
All this code is going to be really hard to maintain in the future. I kind of wish you had used a small library for this. Furthermore, please stop using `var` over `const`, among a few [reasons](https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75), it can leak some memory over the use of multiple functions.

Reply:
This code, as well as well as the one above and directly below (as far as comments) were already on the website, I didn't really look at it since it worked. But yea I'll change that.

Reply:
I see, I was more concerned if it were newly produced code, but seeing it was already there then let's tackle it later on, no need to overburden ourselves right now with refactoring. Let's tackle this into it's own component at some point.